[Bug]: Cannot connect to helper - helper could not be started

### Version

2.5

### Installation Method

Homebrew (brew install --cask vpn-bypass)

### VPN Type

GlobalProtect (Palo Alto)

### Bug Description

Since today I don't get it working again - tried also 2.4.5
All I could see is:

<img width="532" height="134" alt="Image" src="https://github.com/user-attachments/assets/ac03f2ee-2ad9-46f4-acf1-568e4254fd26" />

retry fails also.

### Expected Behavior

Bypasses should get active instead it shows:

<img width="329" height="306" alt="Image" src="https://github.com/user-attachments/assets/6eeed984-fac0-43b8-8888-70d1c228794d" />

### Steps to Reproduce

I use zScaler

### Logs

```shell
[2026-04-22 09:30:00] [INFO] VPN process hint: Zscaler
[2026-04-22 09:29:30] [INFO] VPN process hint: Zscaler
[2026-04-22 09:29:00] [INFO] VPN process hint: Zscaler
[2026-04-22 09:28:33] [INFO] VPN process hint: Zscaler
[2026-04-22 09:28:30] [INFO] VPN process hint: Zscaler
[2026-04-22 09:28:00] [INFO] VPN process hint: Zscaler
[2026-04-22 09:27:30] [INFO] VPN process hint: Zscaler
[2026-04-22 09:27:00] [INFO] VPN process hint: Zscaler
[2026-04-22 09:26:30] [INFO] VPN process hint: Zscaler
[2026-04-22 09:26:21] [ERROR] Helper not ready: Error: Cannot connect to helper after reinstall. Route application skipped.
[2026-04-22 09:26:04] [INFO] VPN process hint: Zscaler
[2026-04-22 09:26:04] [INFO] Network change detected: connected via en0, en8
[2026-04-22 09:26:03] [INFO] VPN process hint: Zscaler
[2026-04-22 09:26:00] [INFO] Config loaded
```

Log - started via terminal:
```
🔔 First launch - using default notification settings (routes OFF)
🚀 First launch - enabling Launch at Login by default
🔐 Helper not found, attempting install...
🔐 Installing privileged helper...
✅ Helper registered successfully via SMAppService
✅ Notification authorization granted
🔐 Helper XPC connection failed, attempting reinstall...
🔐 Installing privileged helper...
✅ Helper registered successfully via SMAppService
🔐 XPC error during getVersion: Couldn’t communicate with a helper application.
🔐 Helper not found, attempting install...
🔐 Installing privileged helper...
✅ Helper registered successfully via SMAppService
🔐 Helper XPC connection failed, attempting reinstall...
🔐 Installing privileged helper...
✅ Helper registered successfully via SMAppService
🔐 XPC error during getVersion: Couldn’t communicate with a helper application.
```

### Environment

MacOS 26.4, M4 Apple Silicon

### Checklist

- [x] I have searched existing issues to ensure this isn't a duplicate
- [x] I have tried restarting the app

--- a/Sources/HelperManager.swift
+++ b/Sources/HelperManager.swift
@@ -259,6 +259,7 @@ final class HelperManager: ObservableObject {
         // binary. For updates we must always go through the legacy path which does
         // the actual file copy. Only use SMAppService for first-time installs.
         let helperPath = "/Library/PrivilegedHelperTools/\(kHelperToolMachServiceName)"
+        let plistPath = "/Library/LaunchDaemons/\(kHelperToolMachServiceName).plist"
         if FileManager.default.fileExists(atPath: helperPath) {
             print("🔐 Helper exists on disk, using legacy path for update...")
             return installHelperLegacy()
@@ -271,10 +272,11 @@ final class HelperManager: ObservableObject {
             print("✅ Helper registered successfully via SMAppService")
 
             // SMAppService.register() can report success without actually placing the
-            // binary on disk (observed with enterprise security tools like Zscaler).
-            // Verify the binary exists before trusting the registration.
-            if !FileManager.default.fileExists(atPath: helperPath) {
-                print("⚠️ SMAppService reported success but helper binary not on disk, falling back to legacy install...")
+            // binary/plist on disk (observed with enterprise security tools like Zscaler).
+            // Verify both files exist before trusting the registration.
+            if !FileManager.default.fileExists(atPath: helperPath) ||
+               !FileManager.default.fileExists(atPath: plistPath) {
+                print("⚠️ SMAppService reported success but helper files missing on disk, falling back to legacy install...")
                 return installHelperLegacy()
             }
 

--- a/Sources/HelperManager.swift
+++ b/Sources/HelperManager.swift
@@ -268,12 +268,21 @@ final class HelperManager: ObservableObject {
             let service = SMAppService.daemon(plistName: "\(kHelperToolMachServiceName).plist")
             try await service.register()
 
-            installationError = nil
             print("✅ Helper registered successfully via SMAppService")
+
+            // SMAppService.register() can report success without actually placing the
+            // binary on disk (observed with enterprise security tools like Zscaler).
+            // Verify the binary exists before trusting the registration.
+            if !FileManager.default.fileExists(atPath: helperPath) {
+                print("⚠️ SMAppService reported success but helper binary not on disk, falling back to legacy install...")
+                return installHelperLegacy()
+            }
+
+            installationError = nil
             return true
         } catch {
             print("⚠️ SMAppService failed: \(error.localizedDescription)")
-            print("🔐 Falling back to legacy SMJobBless...")
+            print("🔐 Falling back to legacy install...")
             return installHelperLegacy()
         }
     }

